panel/lxqtpanel.cpp

```diff
@@ -677,47 +677,64 @@ void LXQtPanel::updateWmStrut()
 
 
 /************************************************
-  The panel can be placed only at the edges of the virtual screen.
-  This function checks if the panel can be placed on the display
-  @screenNum on @position.
+  This function checks if the panel can be placed on
+  the display @screenNum at @position.
+  NOTE: The panel can be placed only at screen edges
+  but no part of it should be between two screens.
  ************************************************/
 bool LXQtPanel::canPlacedOn(int screenNum, LXQtPanel::Position position)
 {
     const auto screens = QApplication::screens();
     if (screens.size() > screenNum)
     {
-        const QRect screenGeometry = screens.at(screenNum)->geometry();
+        QRect screenGeometry = screens.at(screenNum)->geometry();
         switch (position)
         {
         case LXQtPanel::PositionTop:
             for (const auto& screen : screens)
             {
                 if (screen->geometry().top() < screenGeometry.top())
-                    return false;
+                {
+                    screenGeometry.setTop(screen->geometry().top());
```

Comment:
Shouldn't the `screenGeometry` be `const` and the `set* ()` transformations be done on a copy? Modifying the "original" geometry in this loop can misinterpret the next conditions, not?

Reply:
Yes. A silly mistake of mine.

Reply:
> A silly mistake of mine.

But no one would ever be affected until he/she has many monitors arranged in a rectangle :)

Reply:
Done by using `QRect::adjusted()`.

Please check it again. Apparently, I can't see obvious mistakes in this case.